File: src/main/java/com/godwin/jsonparser/ui/tab/IParserTabs.java
package com.godwin.jsonparser.ui.tab;

import com.intellij.ui.tabs.TabInfo;
import com.intellij.ui.tabs.impl.JBEditorTabs;

import javax.swing.*;

/**
 * Created by fingerart on 17/2/27.
 */
public interface IParserTabs {
    IParserTabs addListener(ParserTabsIml.DebuggerTabListener listener);

    IParserTabs addTab(JComponent component, String name);

    int getTabCount();

    TabInfo getTabAt(int i);

    IParserTabs closeTab(int index);

    IParserTabs closeCurrentTab();

    JBEditorTabs getComponent();

    TabInfo getCurrentTab();

    String getTitleAt(int i);

    JComponent getCurrentComponent();
}

File: src/main/kotlin/com/godwin/jsonparser/ui/tabs/IParserTabs.kt
package com.godwin.jsonparser.ui.tabs

import com.intellij.ui.tabs.TabInfo
import com.intellij.ui.tabs.impl.JBEditorTabs
import javax.swing.JComponent

interface IParserTabs {
    fun addListener(listener: ParserTabsImpl.DebuggerTabListener): IParserTabs
    fun addTab(component: JComponent, name: String): IParserTabs
    fun getTabCount(): Int
    fun getTabAt(i: Int): TabInfo
    fun closeTab(index: Int): IParserTabs
    fun closeCurrentTab(): IParserTabs
    fun getComponent(): JBEditorTabs
    fun getCurrentTab(): TabInfo?
    fun getTitleAt(i: Int): String
    fun getCurrentComponent(): JComponent?
}
